[BUG] Redis fails to compile if HAVE_MALLOC_SIZE is not defined

**Describe the bug**

Compiling Redis 6.2.0 on a system without `malloc_usable_size()` or similar with `USE_JEMALLOC=No` results in a build failure. The following is seen on OpenBSD -current (amd64 with clang 10.0.1):

```
zmalloc.c:60:5: error: function-like macro 'sizeof' is not defined
#if PREFIX_SIZE > 0
    ^
zmalloc.c:56:22: note: expanded from macro 'PREFIX_SIZE'
#define PREFIX_SIZE (sizeof(size_t))
                     ^
1 error generated.
```

https://github.com/redis/redis/blob/8e83bcd2acb18370e2d6cea3718339792322e80f/src/zmalloc.c#L60-L64

The issue is that at the preprocessing stage keywords like sizeof do not exist (see C11 6.10, footnote 166), so if HAVE_MALLOC_SIZE is not defined, the above conditional won't be accepted by either clang or gcc.

https://github.com/redis/redis/blob/8e83bcd2acb18370e2d6cea3718339792322e80f/src/zmalloc.c#L50-L58

**To reproduce**

Compile Redis 6.2.0 on a system without `malloc_usable_size()` or similar with `USE_JEMALLOC=No`.

**Expected behavior**

Redis compiles

**Additional information**

I think the following diff would make sense, which I can submit as a PR if that's acceptable.
I expect all versions having a version of https://github.com/redis/redis/pull/8522 to be affected.

```diff
--- src/zmalloc.c.orig
+++ src/zmalloc.c
@@ -28,6 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */

+#include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
@@ -57,7 +58,7 @@ void zlibc_free(void *ptr) {
 #endif
 #endif

-#if PREFIX_SIZE > 0
+#ifndef HAVE_MALLOC_SIZE
 #define ASSERT_NO_SIZE_OVERFLOW(sz) assert((sz) + PREFIX_SIZE > (sz))
 #else
 #define ASSERT_NO_SIZE_OVERFLOW(sz)
```

--- a/.github/workflows/daily.yml
+++ b/.github/workflows/daily.yml
@@ -222,7 +222,7 @@ jobs:
           ./runtest-sentinel &&
           ./runtest-cluster
 
-  test-alpine:
+  test-alpine-jemalloc:
     runs-on: ubuntu-latest
     container: alpine:latest
     steps:
@@ -231,6 +231,26 @@ jobs:
       run: |
           apk add build-base
           make REDIS_CFLAGS='-Werror'
+    - name: test
+      run: |
+        apk add tcl procps
+        ./runtest --accurate --verbose --dump-logs
+    - name: module api test
+      run: ./runtest-moduleapi --verbose
+    - name: sentinel tests
+      run: ./runtest-sentinel
+    - name: cluster tests
+      run: ./runtest-cluster
+
+  test-alpine-libc-malloc:
+    runs-on: ubuntu-latest
+    container: alpine:latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: make
+      run: |
+          apk add build-base
+          make REDIS_CFLAGS='-Werror' USE_JEMALLOC=no
     - name: test
       run: |
         apk add tcl procps

--- a/src/zmalloc.c
+++ b/src/zmalloc.c
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <unistd.h>
+#include <assert.h>
 
 /* This function provide us access to the original libc free(). This is useful
  * for instance to free results obtained by backtrace_symbols(). We need
@@ -49,18 +50,14 @@ void zlibc_free(void *ptr) {
 
 #ifdef HAVE_MALLOC_SIZE
 #define PREFIX_SIZE (0)
+#define ASSERT_NO_SIZE_OVERFLOW(sz)
 #else
 #if defined(__sun) || defined(__sparc) || defined(__sparc__)
 #define PREFIX_SIZE (sizeof(long long))
 #else
 #define PREFIX_SIZE (sizeof(size_t))
 #endif
-#endif
-
-#if PREFIX_SIZE > 0
 #define ASSERT_NO_SIZE_OVERFLOW(sz) assert((sz) + PREFIX_SIZE > (sz))
-#else
-#define ASSERT_NO_SIZE_OVERFLOW(sz)
 #endif
 
 /* Explicitly override malloc/free etc when using tcmalloc. */

